crates/biome_graphql_parser/src/parser/definitions/interface.rs

```diff
@@ -78,6 +78,7 @@ pub(super) fn parse_implements_interface(p: &mut GraphqlParser) -> ParsedSyntax
     let m = p.start();
 
     p.bump(T![implements]);
+    p.eat(T![&]); // leading ampersand separator is optional
```

Comment:
ImplementsInterfaceList allows having leading separator ([here](https://github.com/biomejs/biome/blob/3f311fee9b989244d2be3269664a780cdc4799fb/crates/biome_graphql_parser/src/parser/definitions/interface.rs#L128)), so this should not be necessary?

Reply:
Ah, my bad, I didn't notice the missing element.
I guess the better approach here would be to remove the logic from `parse_list`? It might become too complicated if we try to move `start_list` around 🤔

Reply:
Sure, it seems reasonable because we use the same logic for TypeScript types.